engine/eConstr.ml

```diff
@@ -645,11 +645,19 @@ let eq_constr_universes_proj env sigma m n =
     let res = eq_constr' (unsafe_to_constr m) (unsafe_to_constr n) in
     if res then Some !cstrs else None
 
-let universes_of_constr sigma c =
+let universes_of_constr env sigma c =
   let open Univ in
   let rec aux s c =
     match kind sigma c with
-    | Const (_, u) | Ind (_, u) | Construct (_, u) ->
+    | Const (c, u) ->
+      let open Declarations in
+      begin match (Environ.lookup_constant c env).const_universes with
+        | Polymorphic_const _ ->
+          LSet.fold LSet.add (Instance.levels (EInstance.kind sigma u)) s
+        | Monomorphic_const (univs, _) ->
+          LSet.union s univs
+      end
+    | Ind (_, u) | Construct (_, u) ->
```

Comment:
Are we sure we don't need to do the same for the inductives here?

Reply:
I don't know if there's currently a possibility of building an inductive by tactic to make Coq fail if we don't, but we should do it for sanity's sake.

Reply:
Fixed.